configuration/esapi/ESAPI.properties

```diff
@@ -499,3 +499,13 @@ Validator.DirectoryName=^[a-zA-Z0-9:/\\\\!@#$%^&{}\\[\\]()_+\\-=,.~'` ]{1,255}$
 # Validation of dates. Controls whether or not 'lenient' dates are accepted.
 # See DataFormat.setLenient(boolean flag) for further details.
 Validator.AcceptLenientDates=false
+
+# This is a workaround to make sure that a commit to address GitHub issue #509
+# doesn't accidentally break someone's production code. So essentially what we
+# are doing is to default back to the previous buggy (but, likely by now, expected)
+# behavior. More defaults about this are written up under GitHub issue #521 and
+# the pull request it references. Future major releases of ESAPI (e.g., ESAPI 3.x)
+# will not support this previous buggy behavior, but it will remain for ESAPI 2.x.
+# Set this to 'false' if you want the originally intended behavior of throwing
+# that was fixed via issue #509.
+Validator.ValidationRule.getValid.ignore509Fix=true
```

Comment:
I think the documentation and the implementation are not aligned.  If this is "true" then the implementation logs and returns the empty string.  If  the value is false, then the implementation adheres to the API and docs and throws the expected Exception.

Reply:
Yeah, the semantics here are a little weird.  And I've in multiple instances, run into cases where variable names regarding booleans can lead us down faulty paths.

Reply:
I partly think that the semantics are weird here because, let's face it, this property name sucks dirt. Essentially what the property should be (but in my opinion is even worse) would be something like:
`Validator.ValidationRule.getValid.have_esapi_act_like_it_used_to`
or
`Validator.ValidationRule.getValid.ensure_backward_compatibility_even_though_that_probably_was_a_bug`
which which case 'true' would make more semantic sense.  I think the 'ignore' part of the property name along with sets up a mental dichotomy because it sort of implies a negation that leads to confusion. I think the only way to clear it up may be to come up with a better property name.

So what is a more suitable property name that will cause less confusion? I'm open to suggestions.

@jeremiahjstacey wrote:

> The value of false allows the implementation to act like the other validation rules, and throw an exception if errors are found while validating incoming HTML content.

I think there is one major difference between this and the other rules for validating HTML content and that is those other rules generally don't try to "cleanse" or "sanitize" the output, whereas AntiSamy does. And that's another thing that gives me a little bit of uncertainty about whether the original Javadoc was correct or not.

Reply:
@jeremiahjstacey, @xeno6696  - Still looking for better suggestion for this property name. I can easily rewrite the code, but this is what is most visible to ESAPI users and we need to get this right before we do a release. And I'd like to get some urgency around this PR because this will fix the broken tests in our test suite which we need if we are going to get issue #408 closed out that @kerberosmansour is working on.

I really don't like this property name--at all--but I didn't want it to hold me up from finishing the rest. If we can come up with a name that doesn't imply a double negative, I will gladly adjust the code to these suggestions that you had.